### ISSUE TITLE
Think about what API calls to allow in what circumstances

### ISSUE THREAD

**Issue by alaric-dotmesh** [dotmesh-io/dotmesh#179]
Think properly about which API methods should work only for the admin user or when run in "Hub mode", for security.

See https://docs.google.com/spreadsheets/d/1iSc8hMixsKUhnZ2oZ0eKLH28SLsJWuyaQo1FwoBBsak/edit#gid=0 for a proposed breakdown of API method access rules.

**Comment by prisamuel** [comment]
For example, procure could be restricted to only admin users.

**Comment by alaric-datamesh** [comment]
I've had to leave the `Commit` API open, because the frontend tests do something dodgy that depends on it in order to set up test state. So this issue is done apart from `Commit`, which it's blocked on UI test refactoring for. Pushing it back to the sprint.

**Comment by alaric-dotmesh** [comment]
This is almost a duplicate of https://github.com/dotmesh-io/dotmesh/issues/577
